### PR TITLE
chore(rln-relay): add chain-id flag to wakunode and restrict usage if mismatches rpc provider

### PR DESCRIPTION
**Pull request by rymnc**
# Description

As requested by @alrevuelta to restrict chainId usage and waku node initialization should fail if user erroneously passes in an rpc url for another chain.

# Changes

- [x] All config changes from WakuNodeConf down to GroupManager
- [x] Tests

### CODE REVIEW

File: waku/factory/external_config.nim [hunk 1]
```diff
@@ -76,6 +76,12 @@ type WakuNodeConf* = object
     name: "rln-relay-eth-contract-address"
   .}: string
 
+  rlnRelayChainId* {.
+    desc: "Chain ID of the provided contract (optional, will fetch from RPC provider if not used)",
+    defaultValue: 0,
```
Comment (alrevuelta) [inline waku/factory/external_config.nim:81]: So if not provided it will default to `0` but technically that can be a valid value?

**Review by alrevuelta — COMMENTED** [review]
Not sure i follow. Shouldn't this fail? `rln-relay-chain-id` is wrong

```
docker run quay.io/wakuorg/nwaku-pr:2858 \
--cluster-id=1 \
--rln-relay-eth-client-address=https://rpc.cardona.zkevm-rpc.com \
--log-level=DEBUG \
--rln-relay-chain-id=111111
```

File: waku/factory/external_config.nim (hunk 1 above)
Comment (rymnc) [reply waku/factory/external_config.nim:81]: 0 is an invalid value for chainId. avoided using an `Optio n` here because of that

**Comment by rymnc** [comment]
> Not sure i follow. Shouldn't this fail? `rln-relay-chain-id` is wrong
> 
> ```
> docker run quay.io/wakuorg/nwaku-pr:2858 \
> --cluster-id=1 \
> --rln-relay-eth-client-address=https://rpc.cardona.zkevm-rpc.com \
> --log-level=DEBUG \
> --rln-relay-chain-id=111111
> ```

since you pass in the cluster-id as 1, we override the rln-relay-chain-id, shouldn't that be the intended behaviour?

**Comment by alrevuelta** [comment]
> since you pass in the cluster-id as 1,

indeed, my bad.

**Pull request merged**